tst/EnergyPlus/unit/EconomicTariff.unit.cc

```diff
@@ -677,3 +677,297 @@ TEST_F(EnergyPlusFixture, EconomicTariff_GatherForEconomics)
+    EXPECT_EQ(state->dataEconTariff->econVar(1).values(6), 6.00);
+    EXPECT_EQ(state->dataEconTariff->econVar(1).values(7), 7.00);
+    EXPECT_EQ(state->dataEconTariff->econVar(1).values(8), 8.00);
+    EXPECT_EQ(state->dataEconTariff->econVar(1).values(9), 9.00);
+    EXPECT_EQ(state->dataEconTariff->econVar(1).values(10), 10.00);
+    EXPECT_EQ(state->dataEconTariff->econVar(1).values(11), 11.00);
+    EXPECT_EQ(state->dataEconTariff->econVar(1).values(12), 12.00);
+
+    std::string const idf_objects1 = delimited_string({
+
+        "Schedule:Compact,",
+        "  Electricity Season Schedule,  !- Name",
+        "  Any Number,              !- Schedule Type Limits Name",
+        "  Through: 5/31,           !- Field 1",
+        "  For: AllDays,            !- Field 2",
+        "  Until: 24:00,            !- Field 3",
+        "  1,                       !- Field 4",
+        "  Through: 9/30,           !- Field 5",
+        "  For: AllDays,            !- Field 6",
+        "  Until: 24:00,            !- Field 7",
+        "  3,                       !- Field 8",
+        "  Through: 12/31,          !- Field 9",
+        "  For: AllDays,            !- Field 10",
+        "  Until: 24:00,            !- Field 11",
+        "  1;                       !- Field 12",
+
+        "UtilityCost:Tariff,",
+        "  ExampleAWithVariableMonthlyCharge,     !- Name",
+        "  ElectricityNet:Facility, !- Output Meter Name",
+        "  kWh,                     !- Conversion Factor Choice",
+        "  ,                        !- Energy Conversion Factor",
+        "  ,                        !- Demand Conversion Factor",
+        "  ,                        !- Time of Use Period Schedule Name",
+        "  Electricity Season Schedule,  !- Season Schedule Name",
+        "  ,                        !- Month Schedule Name",
+        "  ,                        !- Demand Window Length",
+        "  0,                       !- Monthly Charge or Variable Name",
+        "  ,                        !- Minimum Monthly Charge or Variable Name",
+        "  ,                        !- Real Time Pricing Charge Schedule Name",
+        "  ,                        !- Customer Baseline Load Schedule Name",
+        "  ,                        !- Group Name",
+        "  NetMetering;             !- Buy Or Sell",
+
+        "UtilityCost:Variable,",
+        "VariableFixedCharge, !-Name",
+        "ExampleAWithVariableMonthlyCharge, !-Tariff Name",
+        "Energy, !-Variable Type",
+        "1.00, !-January Value",
+        "2.00, !-February Value",
+        "3.00, !-March Value",
+        "4.00, !-April Value",
+        "5.00, !-May Value",
+        "6.00, !-June Value",
+        "7.00, !-July Value",
+        "8.00, !-August Value",
+        "9.00, !-September Value",
+        "10.00, !-October Value",
+        "11.00, !-November Value",
+        "12.00; !-December Value"});
+
+    ASSERT_TRUE(process_idf(idf_objects1));
```

Comment:
Hmm, processing multiple IDF snippets in one unit test.  Seems...susceptible to problems later.  That's equivalent to asking EnergyPlus to process two different IDFs.  I would like to see this simply split into different unit tests to ensure they don't accidentally cause problems later.

Reply:
The unit tests are now divided into individual ones. Thanks!